libfreerdp/core/capabilities.c

```diff
@@ -2137,11 +2139,30 @@ static BOOL rdp_apply_virtual_channel_capability_set(rdpSettings* settings, cons
 	WINPR_ASSERT(settings);
 	WINPR_ASSERT(src);
 
-	if (!settings->ServerMode)
-	{
-		settings->VirtualChannelCompressionFlags = src->VirtualChannelCompressionFlags;
+	/* MS servers and clients disregard in advertising what is relevant for their own side */
+	if (settings->ServerMode && settings->VirtualChannelCompressionFlags & VCCAPS_COMPR_SC &&
```

Comment:
some braces might help here to make the masks and conditions easier to read

Reply:
Done.